icu4c/source/test/intltest/formatted_string_builder_test.cpp

```diff
@@ -173,25 +192,28 @@ void NumberStringBuilderTest::testCopy() {
     }
 }
 
-void NumberStringBuilderTest::testFields() {
+void FormattedStringBuilderTest::testFields() {
     UErrorCode status = U_ZERO_ERROR;
     // Note: This is a C++11 for loop that calls the UnicodeString constructor on each iteration.
     for (UnicodeString str : EXAMPLE_STRINGS) {
-        NumberStringBuilder sb;
+        FormattedValueStringBuilderImpl sbi(0);
+        FormattedStringBuilder& sb = sbi.getStringRef();
         sb.append(str, UNUM_FIELD_COUNT, status);
         assertSuccess("Appending to sb", status);
         sb.append(str, UNUM_CURRENCY_FIELD, status);
         assertSuccess("Appending to sb", status);
         assertEquals("Reference string copied twice", str.length() * 2, sb.length());
         for (int32_t i = 0; i < str.length(); i++) {
-            assertEquals("Null field first", (Field) UNUM_FIELD_COUNT, sb.fieldAt(i));
-            assertEquals("Currency field second", (Field) UNUM_CURRENCY_FIELD, sb.fieldAt(i + str.length()));
+            assertEquals("Null field first",
+                (FormattedStringBuilder::Field) UNUM_FIELD_COUNT, sb.fieldAt(i));
+            assertEquals("Currency field second",
+                (FormattedStringBuilder::Field) UNUM_CURRENCY_FIELD, sb.fieldAt(i + str.length()));
         }
 
         // Very basic FieldPosition test. More robust tests happen in NumberFormatTest.
         // Let NumberFormatTest also take care of FieldPositionIterator material.
         FieldPosition fp(UNUM_CURRENCY_FIELD);
-        sb.nextFieldPosition(fp, status);
+        sbi.nextFieldPosition(fp, status);
```

Comment:
Why did this one change from sb to sbi?

Reply:
I moved the method `nextFieldPosition` from NumberStringBuilder to FormattedValueStringBuilderImpl instead of to FormattedStringBuilder because the implementation of `nextFieldPosition` requires the same plumbing as `nextPosition`, which I moved to that new class in order to put the FormattedValue methods in one place separate from FormattedStringBuilder.